app/models/course/registration.rb

```diff
@@ -0,0 +1,61 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: [roles[:teaching_assistant], roles[:manager], roles[:owner]]) }
+
+  enum role: CourseUser.roles
+
+  belongs_to :course
+  belongs_to :user
+
+  def staff?
+    teaching_assistant? || manager? || owner?
+  end
+
+  # Approve the registration record by creating a CourseUser of the user to the course with the
+  # requested role.
+  def approve
+    if course.course_users.create(user_id: user_id, role: role)
+      true
+    else
+      course.course_user.errors.each do |attr, message|
+        errors[attr] = message
+      end
+      false
+    end
+  end
+
+  # Process the registration records of given IDs and then delete them.
+  #
+  # @param [Course] the course that the registration records belong to.
+  # @param [Array<Int>] the IDs of registration records
+  # @options approve: If specified a true equivalent value, the Registration#approve method will
+  #                   be called. Default to false.
+  #          process_all: if specified a true equivalent value, all registration to the course
+  #                       will be processed. Default to false.
+  #          role: If specified a value in the Registration.roles enum, only registrations of that
+  #                role will be processed. Other IDs are ignored. Default to nil.
+  # @return [Hash] If successful, the hash contains an array of processed IDs with the key -
+  #                'processed_ids'. Otherwise it contains an ActiveModel::Errors with the key -
+  #                'errors'.
+  def self.process(course, registration_ids, options = {})
```

Comment:
Static methods are not good for this purpose.

I also do not think this function is good, abstraction wise. Do not pass IDs around because that breaks abstraction barriers.